cpp/core/jni/JniWrapper.cc

```diff
@@ -564,6 +565,39 @@ Java_io_glutenproject_vectorized_NativeColumnarToRowJniWrapper_nativeClose(JNIEn
   JNI_METHOD_END()
 }
 
+JNIEXPORT void JNICALL Java_io_glutenproject_vectorized_NativeRowToColumnarJniWrapper_nativeConvertRowToColumnar(
```

Comment:
You can return `ArrowCStructColumnarBatch` from c++ to Java to avoid assembling Arrow vectors from Java side.

I'll open a issue for that.

Reply:
https://github.com/oap-project/gluten/issues/1120